remme/rpc_api/event.py

```diff
@@ -0,0 +1,458 @@
+        message_content=req_msg.SerializeToString(),
+        timeout=ZMQ_CONNECTION_TIMEOUT)
+
+    LOGGER.debug(f'Message type: {msg.message_type}')
+
+    # Validate the response type
+    if msg.message_type != Message.CLIENT_EVENTS_SUBSCRIBE_RESPONSE:
+        raise ClientException(
+            message=f'Unexpected message type {msg.message_type}')
+
+    # Parse the response
+    response = ClientEventsSubscribeResponse()
+    response.ParseFromString(msg.content)
+
+    # Validate the response status
+    if response.status != ClientEventsSubscribeResponse.OK:
+        if response.status == ClientEventsSubscribeResponse.UNKNOWN_BLOCK:
+            raise ClientException(
+                message=f'Unknown block "{from_block}"')
+        raise ClientException(
+            message='Subscription failed: Couldn\'t '
+                    'send multipart')
+
+    if ws not in request.rpc._evthashes:
+        request.rpc._evthashes[ws] = set()
+
+    subsevt[event_type] = {
+        'msg_id': msg_id,
+        'validated_data': validated_data,
+    }
+
+    request.rpc.loop.create_task(_consumer(request))
+    if event_type == 'batch':
+        request.rpc.loop.create_task(_producer(request))
+
+    return 'SUBSCRIBED'
+
+
+async def unsubscribe(request):
+    ws = request.ws
+    if ws is None:
+        raise ClientException(
+            message='Subscription available only through websocket')
+
+    request.params = request.params or {}
+    try:
+        event_type = request.params['event_type']
+    except KeyError as e:
+        raise RpcInvalidParamsError(message='Missed event_type')
+
+    subsevt = request.rpc._subsevt.get(ws, {})
+    try:
+        del subsevt[event_type]
+    except KeyError:
+        raise ClientException(
+            message='Subscription not found')
+
+    return 'UNSUBSCRIBED'
+
+
+async def _consumer(request):
+    ws = request.ws
+    stream = ws.stream
+
+    while not ws.closed:
+        LOGGER.debug(f'Consumer: Start fetching a new messages for {ws}...')
+        datas = []
+        try:
+            msg = await stream.receive()
+        except asyncio.QueueEmpty:
+            continue
+        else:
+            LOGGER.debug('Msg received for zmq socket '
+                         f'{stream._socket}')
+            datas.append((request, msg))
+
+        await asyncio.gather(*(_process_msg(*data) for data in datas))
+        LOGGER.debug(f'Consumer: Waiting new messages for {ws}...')
+
+
+async def _producer(request):
+    ws = request.ws
+    stream = ws.stream
+
+    while not ws.closed:
+        LOGGER.debug('Producer: Start producing a new messages for {ws}...')
+        for evt_name, data in request.subsevt.get(ws, {}).items():
+            evt_tr = EVENT_TRANSLATOR[evt_name]
+            await evt_tr.produce_custom_msg(stream, data['validated_data'])
+        LOGGER.debug('Producer: Start producing a new messages for {ws}...')
+        await asyncio.sleep(1)
```

Comment:
We can just wait for `block-commit` here. Rationale: batch status usually doesn't change until a new chain head arrival.